QappTuning.py

```diff
@@ -0,0 +1,585 @@
+ref_w = 120
+ref_h = 80
+rc1 = (0, 0)
+rc2 = (0, ref_h)
+rc3 = (ref_w, ref_h)
+rc4 = (ref_w, 0)
+ref_corns = np.array((rc1, rc2, rc3, rc4), np.float32)
+ref_data = (ref, ref_w, ref_h, ref_corns)
+
+# function that checks for macbeth close to camera
+
+
+def my_find_macbeth(img):
+    warnings.simplefilter("ignore")
+    fxn()
+    cor, mac, coords, msg = get_macbeth_chart(img, ref_data)
+    if cor < 0.75:
+        a = 2
+        img_br = cv2.convertScaleAbs(img, alpha=a, beta=0)
+        cor_b, mac_b, coords_b, msg_b = get_macbeth_chart(img_br, ref_data)
+        if cor_b > cor:
+            cor, _, coords, _ = cor_b, mac_b, coords_b, msg_b
+    if cor < 0.75:
+        a = 4
+        img_br = cv2.convertScaleAbs(img, alpha=a, beta=0)
+        cor_b, mac_b, coords_b, msg_b = get_macbeth_chart(img_br, ref_data)
+        if cor_b > cor:
+            cor, _, coords, _ = cor_b, mac_b, coords_b, msg_b
+    if cor > 0.5:
+        return (cor, coords)
+    else:
+        return (0, None)
+
+# future parameters
+
+
+pool = ThreadPoolExecutor(max_workers=4)
+overlay_active = False
+busy = False
+
+# Class that makes the wisget itself that will enclose macbeth if it sees it
+
+
+class MacbethWindow(QWidget):
+    done_signal = pyqtSignal()
+
+    def __init__(self, *args, **kwargs):
+        super(QWidget, self).__init__(*args, **kwargs)
+        self.qpicamera2 = QGlPicamera2(picam2, width=800, height=600, keep_ar=True, bg_colour=(236, 236, 236))
+        self.qpicamera2.done_signal.connect(self.capture_done)
+
+        self.window = QWidget()
+        self.layout_h = QHBoxLayout()
+        self.layout_h.addWidget(self.qpicamera2, 80)
+        self.window.resize(1200, 600)
+        self.window.setLayout(self.layout_h)
+
+        self._timer = QTimer()
+        self._timer.setInterval(20)
+        self._timer.timeout.connect(self.on_button_clicked)
+        self._timer.start()
+        picam2.start()
+
+        self.count = 0
+        self.futures = []
+        self.filter_machb = 0
+
+    def on_button_clicked(self):
+        if busy is False:
+            picam2.capture_array(signal_function=self.qpicamera2.signal_done)
+
+    def signal_done(self, job):
+        self.done_signal.emit()
+
+    def capture_done(self, job):
+        global overlay_active
+        array = job.get_result()
+        if (self.count < 1):
+            # add another future
+            future = pool.submit(my_find_macbeth, array)
+            self.futures.append(future)
+            self.count += 1
+
+        if (self.futures != []):
+            if self.futures[0].running() is False:
+                cor, macbeth = self.futures[0].result()
+                self.futures.pop(0)
+                self.count -= 1
+                # OVERLAY
+                if (cor != 0):
+                    overlay_active = True
+                    self.filter_machb = 0
+                    h = array.shape[0]
+                    w = array.shape[1]
```

Comment:
Pythonistas might prefer
```
h, w = array.shape
```

Reply:
There are more than two parameters in array.shape